### ISSUE TITLE
[no-unsafe-member-access] false positive?

### ISSUE THREAD

**Issue by OmgImAlexis** [typescript-eslint/typescript-eslint#1878]
**What were you trying to do?**

```json
{
  "rules": {
    "@typescript-eslint/no-unsafe-member-access": ["error"]
  }
}
```

This is the `docker` file.
```ts
import pify from 'pify';
import Docker from 'dockerode';
import { paths } from '../../paths';

// Borrowed from https://stackoverflow.com/a/52731696 until pify
// adds their own types, check https://github.com/sindresorhus/pify/issues/74
type UnpackedPromise<T> = T extends Promise<infer U> ? U : T
type GenericFunction<TS extends any[], R> = (...args: TS) => R
type Promisify<T> = {
	[K in keyof T]: T[K] extends GenericFunction<infer TS, infer R>
	? (...args: TS) => Promise<UnpackedPromise<R>>
	: never
}

const socketPath = paths.get('docker-socket') ?? '/var/run/docker.sock';
const client = new Docker({
	socketPath
});
const promisifiedClient = pify(client);

/**
 * Docker client
 */
export const docker = promisifiedClient as unknown as Promisify<typeof client>;
```

In another file I have a piece of code like so. I've added what intellisense shows.
```ts
const installed = await docker // (alias) const docker: Promisify<Docker>
		.listContainers({ all: true }) // (method) listContainers(options?: {} | undefined): Promise<Dockerode.ContainerInfo[]>
```

**What did you expect to happen?**
No error.

**What actually happened?**
```
Unsafe member access .listContainers on an any value  @typescript-eslint/no-unsafe-member-access
```
**Versions**

| package                                 | version |
| --------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`      | `2.26.0` |
| `@typescript-eslint/parser`             | `2.26.0 ` |
| `@typescript-eslint/typescript-estree`  | `2.26.0` |
| `@typescript-eslint/experimental-utils` | `2.26.0` |
| `TypeScript`                            | `3.8.3` |
| `node`                                  | `v12.5.0` |
| `npm`                                   | `6.13.0` |

**Comment by bradzacher** [comment]
Could you please provide a repro repo?
It's hard to investigate this due to the number of module types and files involved.

**Comment by OmgImAlexis** [comment]
Found the issue.
